website/addons/evernote/views.py

```diff
@@ -0,0 +1,215 @@
+
+    uid = folder['id']
+    path = folder['path']
+
+    node_addon.set_folder(uid, auth=auth)
+
+    return {
+        'result': {
+            'folder': {
+                'name': path.replace('All Files', '') if path != 'All Files' else '/ (Full Evernote)',
+                'path': path,
+            },
+            'urls': serializer.addon_serialized_urls,
+        },
+        'message': 'Successfully updated settings.',
+    }
+
+
+@must_have_addon('evernote', 'user')
+@must_have_addon('evernote', 'node')
+@must_have_permission(permissions.WRITE)
+def evernote_add_user_auth(auth, node_addon, user_addon, **kwargs):
+    """Import evernote credentials from the currently logged-in user to a node.
+    """
+    external_account = ExternalAccount.load(
+        request.json['external_account_id']
+    )
+
+    if external_account not in user_addon.external_accounts:
+        raise HTTPError(http.FORBIDDEN)
+
+    try:
+        node_addon.set_auth(external_account, user_addon.owner)
+    except PermissionsError:
+        raise HTTPError(http.FORBIDDEN)
+
+    node_addon.set_user_auth(user_addon)
+    node_addon.save()
+
+    return {
+        'result': EvernoteSerializer().serialize_settings(node_addon, auth.user),
+        'message': 'Successfully imported access token from profile.',
+    }
+
+
+evernote_deauthorize_node = generic_views.deauthorize_node(
+    SHORT_NAME
+)
+
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+def evernote_notes(node_addon, **kwargs):
+
+    token = node_addon.external_account.oauth_key
+    client = utils.get_evernote_client(token)
+
+    # will want to pick up notes for the notebook
+    # start with calculating the number of notes in nb
+
+    notes = utils.notes_metadata(client,
+                    notebookGuid=node_addon.folder_id,
+                    includeTitle=True,
+                    includeUpdated=True,
+                    includeCreated=True)
+
+    results = [{'title': note.title,
+              'guid': note.guid,
+              'created': utils.timestamp_iso(note.created),
+              'updated': utils.timestamp_iso(note.updated)}
+              for note in notes]
+
+    return results
+
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+def evernote_note(node_addon, noteid, **kwargs):
+
+    token = node_addon.external_account.oauth_key
+    client = utils.get_evernote_client(token)
+    note = utils.get_note(client, noteid,
+        withContent=True,
+        withResourcesData=False)
+
+    # convert to HTML
+    mediaStore = utils.MyMediaStore(client.get_note_store(), note.guid)
+    html = ENML2HTML.ENMLToHTML(note.content, pretty=True, header=False,
+          media_store=mediaStore)
+
+    return {'noteid': noteid,
+            'content': note.content,
+            'html': html}
+
+@must_have_addon('evernote', 'node')
+@must_be_addon_authorizer('evernote')
+def evernote_folder_list(node_addon, **kwargs):
```

Comment:
Given that Evernote doesn't have allow nesting with their Notebooks, you can probably remove the `None` case, and do a few overrides in `node-cfg.js` to hide the `+` icons for folder expanding. Alternatively, you could do something like what Dataverse does to select which Dataset should be connected.